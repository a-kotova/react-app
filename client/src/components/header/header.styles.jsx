import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const HeaderContainer = styled.div`
  height: 50px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 0px;
  
  @media screen and (max-width: 800px) {
    height: 60px;
    padding: 10px;
    margin-bottom: 10px;
  }
`;

HeaderContainer.displayName = 'HeaderContainer';

export const LogoContainer = styled(Link)`
  height: 100%;
  width: 50px;
  
  @media screen and (max-width: 800px) {
    width: 50px;
    padding: 0;
  }
`;

LogoContainer.displayName = 'LogoContainer';

export const OptionsContainer = styled.div`
  width: 50%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  
  @media screen and (max-width: 800px) {
    width: 80%;
  }
`;

OptionsContainer.displayName = 'OptionsContainer';

export const OptionLink = styled(Link)`
  padding: 10px 15px;
  cursor: pointer;
`;

OptionLink.displayName = 'OptionLink';
