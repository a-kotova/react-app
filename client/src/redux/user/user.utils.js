export const setCurrentUser = (snapshot) => (
  {
    id: snapshot.id,
    ...snapshot.data,
  }
);
