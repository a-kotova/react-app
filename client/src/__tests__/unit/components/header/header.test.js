import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from '../../../../components/header/header.component';
import { signOutStart } from '../../../../redux/user/user.actions';
import getMockedState from "../../../utils/mock-state-provider";

describe('Components: Header', () => {
  const initialState = getMockedState(['cart', 'user']);

  describe('Header', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <Header
          hidden={initialState.cart.hidden}
          currentUser={initialState.user.currentUser}
        />,
      );
    });

    it('should have logo as a link to homepage', () => {
      expect(wrapper.find('LogoContainer').prop('to')).toEqual('/');
    });

    it('should have link to "Shop"', () => {
      expect(wrapper.find('OptionLink[to="/shop"]').text()).toEqual('SHOP');
    });

    it('should have link to "Contact"', () => {
      expect(wrapper.find('OptionLink[to="/contact"]').text()).toEqual('CONTACT');
    });

    it('should have link to "Sign In" if user is not logged in', () => {
      expect(wrapper.find('OptionLink[to="/signing"]').text()).toEqual('SIGN IN');
    });

    it('should have link to "Sign Out" if user is logged in', () => {
      const currentUser = { id: 'kjfjksfal123', displayName: 'User Name' };
      wrapper.setProps({ currentUser });

      expect(wrapper.find('OptionLink').at(2).text()).toEqual('SIGN OUT');
    });

    it('should be able to log out user', () => {
      const currentUser = { id: 'kjfjksfal123', displayName: 'User Name' };
      const signOutStartMock = jest.fn();
      wrapper.setProps({ currentUser, signOutStart: signOutStartMock });
      wrapper.find('OptionLink').at(2).prop('onClick')();

      expect(signOutStartMock).toHaveBeenCalledTimes(1);
    });

    it('should hide cart dropdown', () => {
      wrapper.setProps({ hidden: true });

      expect(wrapper.find('withRouter(Connect(CartDropdown))')).toHaveLength(0);
    });

    it('should show cart dropdown', () => {
      wrapper.setProps({ hidden: false });

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('withRouter(Connect(CartDropdown))')).toHaveLength(1);
    });
  });

  describe('mapStateToProps', () => {
    it('should map state', () => {
      expect(mapStateToProps(initialState))
        .toContainEntries([['currentUser', initialState.user.currentUser], ['hidden', initialState.cart.hidden]]);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should dispatch actions', () => {
      const dispatch = jest.fn();

      expect(JSON.stringify(mapDispatchToProps(dispatch)))
        .toEqual(JSON.stringify({ signOutStart }));
    });
  });
});
