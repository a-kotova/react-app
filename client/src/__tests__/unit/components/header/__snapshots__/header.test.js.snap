// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Header Header should show cart dropdown 1`] = `
<HeaderContainer>
  <LogoContainer
    to="/"
  >
    <ForwardRef(SvgCrown) />
  </LogoContainer>
  <OptionsContainer>
    <withRouter(Connect(SearchBar)) />
    <OptionLink
      to="/shop"
    >
      SHOP
    </OptionLink>
    <OptionLink
      to="/contact"
    >
      CONTACT
    </OptionLink>
    <OptionLink
      to="/signing"
    >
      SIGN IN
    </OptionLink>
    <Connect(CartIcon) />
  </OptionsContainer>
  <withRouter(Connect(CartDropdown)) />
</HeaderContainer>
`;
